parbase/FairRuntimeDb.cxx

```diff
@@ -363,144 +355,42 @@ Int_t FairRuntimeDb::findOutputVersion(FairParSet* cont)
+    Int_t containerVersion = 0;
     if (getOutput() && output->check() && output->isAutoWritable()) {
-        cout << " CHECK OUTPUT DONE " << endl;
-        if (isRootFileOutput) {
-            if (cont->hasChanged()) {
-                cv = findOutputVersion(cont);
-                if (cv == 0) {
-                    cv = cont->write(output);
-                    if (cv > 0) {
-                        LOG(info) << "***  " << c << " written to ROOT file   version: " << cv;
-                    } else if (cv == -1) {
-                        return kFALSE;
-                    }
-                    // -1 indicates and error during write
-                    // 0 is allowed for all containers which have no write function
-                }
-                vers->setRootVersion(cv);
-            } else {
-                if (vers->getRootVersion() == 0) {
-                    cv = findOutputVersion(cont);
-                    vers->setRootVersion(cv);
-                }
+        const Text_t* outputName = output->GetName();
+        if (container->hasChanged()) {
+            containerVersion = container->write(output);
+            if (containerVersion == -1) {
+                return kFALSE;
             }
-        } else {   // might be Ascii I/O
-            if (cont->hasChanged()) {
-                cv = cont->write(output);
-                if (cv < 0) {
-                    return kFALSE;
-                }
-                cout << "***  " << c << " written to output" << '\n';
-                vers->setRootVersion(cv);
-            }
-        }
-    }
-    vers->setInputVersion(cont->getInputVersion(1), 1);
-    vers->setInputVersion(cont->getInputVersion(2), 2);
-    cont->setChanged(kFALSE);
-    if (refRun) {
-        FairParVersion* refVers = refRun->getParVersion(c);
-        if (refVers) {
-            refVers->setInputVersion(cont->getInputVersion(1), 1);
-            refVers->setInputVersion(cont->getInputVersion(2), 2);
-            refVers->setRootVersion(cv);
-        }
-    }
-    return rc;
-}
-#endif
 
-//////////// With DB modification (FIXME FIXME) //////
-#if (USE_DB_METHOD > 0)
-Bool_t FairRuntimeDb::writeContainer(FairParSet* cont, FairRtdbRun* run, FairRtdbRun* refRun)
-{
-    // std::cout << "\n -I FairRuntimeDB Using DB mode \n";
-    // writes a container to the output if the containers has changed
-    // The output might be suppressed if the changes is due an initialisation from a
-    //   ROOT file which serves also as output or if it was already written
-    const Text_t* c = cont->GetName();
-    LOG(debug) << "RuntimeDb: write container: " << cont->GetName();
-    FairParVersion* vers = run->getParVersion(c);
-    Bool_t rc = kTRUE;
-    Int_t cv = 0;
-    if (getOutput() && output->check() && output->isAutoWritable()) {
-        switch (ioType) {
-            case RootFileOutput:   // RootFile
-                if (cont->hasChanged()) {
-                    cv = findOutputVersion(cont);
```

Comment:
If we remove this, we should make sure, it is no longer needed.

Reply:
I did remove it since (from my point of view) requiring `cont->hasChanged()` and `findOutputVersion(cont) == 0` is the same check.

Reply:
> I did remove it since (from my point of view) requiring `cont->hasChanged()` and `findOutputVersion(cont) == 0` is the same check.

I agree that this is the case, if the `FairParSet` input versions at indices `1` and `2` are `-1`, but what about these cases:
https://github.com/FairRootGroup/FairRoot/blob/0d7a49edf0c4a2fe7cbc247d4e867abb110c4fa1/parbase/FairRuntimeDb.cxx#L350-L362

@fuhlig1 @MohammadAlTurany Do you have more insight here? What are these input versions used for? Why have these cases above been handled for root file and TSQL outputs und not for ASCII file output. Shouldn't they be handled always?

Reply:
As far as I remember ASCII does not have versions, it is only for debugging or overwriting parameters.